hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImageFormatProtobuf.java

```diff
@@ -172,13 +172,55 @@ public LoaderContext getLoaderContext() {
       return ctx;
     }
 
+    /***
+     * a thread for parallel MD5 computing to increase performance when loading
+     */
+    private static class DigestThread extends Thread {
+      volatile private IOException ioe = null;
+      volatile private MD5Hash digest = null;
+      private File file;
+
+      public DigestThread(File inFile) {
+        file = inFile;
```

Comment:
It would be nice to have a name for the thread:
- setName
- toString (and let the daemon library do it if so)